src/compiler/checker.ts

```diff
@@ -1850,7 +1851,8 @@ namespace ts {
             isUse: boolean,
             excludeGlobals: boolean,
             getSpellingSuggestions: boolean,
-            lookup: typeof getSymbol): Symbol | undefined {
+            lookup: typeof getSymbol,
+            reportErrors = true): Symbol | undefined {

```

Comment:
`resolveNameHelper` used to report errors even if `nameNotFoundMessage` was `undefined`. Since we now call `resolveNameHelper` in `getReferencedValueOrAliasSymbol` when skipping the cached resolved symbol, without this change we would get different number of pre-emit and post-emit reported errors, and they'd be useless/repeated errors anyways.

Reply:
`resolveEntityName` already takes an `ignoreErrors` flag and is seemingly what we use to resolve type names elsewhere - it just passes an `undefined` `nameNotFoundMessage` in those cases. Instead of adding a new parameter that also needs to be passed in at the `resolveEntityName` and maybe other callsites, maybe checking the existing `nameNotFoundMessage` parameter in more places works?

Reply:
Isn't the existing code in the checker somehow relying on the fact that we report some errors even when `nameNotFoundMessage` is `undefined`? I thought this was intentional and the check for an undefined `nameNotFoundMessage` just kept us from reporting some "not found" errors.

Reply:
*probably* not. I'd change it and see if it breaks anything.